### PR TITLE
BE-9-route-lincoln-item+

### PR DESCRIPTION
**Pull request by lincoln1883**
Created tests case for all the item routes, to  check  errors and responses.
Created the item routes.

### CODE REVIEW

File: server/src/__tests__/item.test.ts [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+import connection from "../db/connection";
+import app from "../app";
+
+const HOME_ROUTE = "/items";
+let itemId: string;
+const INVALID_UUID = "thisisan-inva-lidu-uidv-aluesoerror";
+const NON_EXISTENT_UUID = "12345678-1234-1234-1234-1234567890AB";
+let supplierId: string;
+
+describe("Operation on the item route", () => {
```
Comment (alcb1310) [inline server/src/__tests__/item.test.ts:12]: There are no tests validating data when user tries to update an item, there should be tests for that. Eg:
- "Should return 400 if user sends incorrect data"
- "should return 200 if user sends correct data"

File: server/src/routes/item.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  getAllItems,
+  getOneItem,
+  updateOneItem,
+} from "../controller/item";
+import { validateUUID } from "../middleware/validateUUID";
+import { itemCreateValidator } from "../validators";
+
+const router: Router = Router();
+
+router.post(
```
Comment (alcb1310) [inline server/src/routes/item.ts:14]: According with swagger, when the user creates a new item, he/she should send an array of tags that are item related, so we will need to validate that we are receiving an array of strings ["tag1", "tag2", "tag3", ..., "tagN"] and the app should know it is a new tag and create the tag and if it exists, it should just have to connect both items

File: server/src/routes/item.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  "/",
```
Comment (alcb1310) [inline server/src/routes/item.ts:15]: How will you propose we handle the adding pictures to the items?

File: server/src/__tests__/item.test.ts (hunk 1 above)
Comment (lincoln1883) [reply server/src/__tests__/item.test.ts:12]: I have test cases for these two points on line 21 & line 86 respectivly.

**Pull request merged**